src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -123,32 +124,6 @@ public partial struct SyntaxValueProvider
         }
     }
 
-    private static ImmutableArray<(SyntaxTree Tree, SourceGeneratorSyntaxTreeInfo Info)> GetSourceGeneratorInfo(
-        ISyntaxHelper syntaxHelper, Compilation compilation, CancellationToken cancellationToken)
-    {
-        // Get the count up front so we can allocate without waste.
-        var count = 0;
-        foreach (var tree in compilation.CommonSyntaxTrees)
-        {
-            cancellationToken.ThrowIfCancellationRequested();
-            var info = tree.GetSourceGeneratorInfo(syntaxHelper, cancellationToken);
-            if ((info & SourceGeneratorSyntaxTreeInfo.ContainsGlobalAliasesOrAttributeList) != 0)
-                count++;
-        }
-
-        var builder = ImmutableArray.CreateBuilder<(SyntaxTree Tree, SourceGeneratorSyntaxTreeInfo Info)>(count);
-
-        // Iterate again.  This will be free as the values from before will already be cached on the syntax tree.
-        foreach (var tree in compilation.CommonSyntaxTrees)
-        {
-            var info = tree.GetSourceGeneratorInfo(syntaxHelper, cancellationToken);
-            if ((info & SourceGeneratorSyntaxTreeInfo.ContainsGlobalAliasesOrAttributeList) != 0)
-                builder.Add((tree, info));
-        }
-
-        return builder.MoveToImmutable();
-    }
-
     private static ImmutableArray<SyntaxNode> GetMatchingNodes(
```

Comment:
Where did this code go?

Reply:
If you mean the removed method, it was only called in `.SelectMany((compilation, cancellationToken) => GetSourceGeneratorInfo(syntaxHelper, compilation, cancellationToken))` above which I replaced by two steps with equivalent behavior. Instead of always getting only syntax trees with attributes or global usings, I use two steps - first get all syntax trees, then filter those with attributes or global usings. See point 2 in PR description on why I think this is better.